gitpod-io/gitpod
# [go] Make go workspace work

## Description
This PR changes the `go.work` file to list all go modules but the caddy plugins and docker-up (they both introduce errors we can look into separately).

Using a go workspace provides `go work sync` (that updates all dependencies) and more importantly helps gopls understand our workspace.

## Related Issue(s)

Fixes #

## How to test

1. start workspace with VS Code
2. verify working on the components listed in `go.work` works nicely. E.g. running tests, diagnostics, etc.

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:

      Valid options are `all`, `workspace`, `webapp`, `ide`
